drop py2 support + convert README to md + travis improvements

jhoblitt opened the pull request.

jhoblitt committed:
drop py2 from travis matrix

jhoblitt committed:
fix flake8 W605 errors

jhoblitt committed:
rm usage of __future__

jhoblitt committed:
convert README -> README.md

jhoblitt committed:
add travis badge

timj commented:
There are some `builtins` imports that also need to be removed.

timj commented:
There are also some flake8 warnings (at least on master):
```
$ flake8
./python/lsst/ci/build.py:190:38: W605 invalid escape sequence '\.'
./python/lsst/ci/prepare.py:116:31: W605 invalid escape sequence '\w'
./python/lsst/ci/prepare.py:644:17: E117 over-indented (comment)
./python/lsst/ci/prepare.py:645:17: E117 over-indented
./python/lsst/ci/prepare.py:646:25: E117 over-indented
./python/lsst/ci/prepare.py:648:33: E117 over-indented
```

timj commented on .travis.yml:
```diff
@@ -14,8 +12,6 @@ matrix:
         python setup.py test
         # https://github.com/tholo/pytest-flake8/issues/27
         flake8 bin/lsst-build
```
Why isn't this checking the other python code?

jhoblitt replied:
see the comment above -- I couldn't figure out how to get it to check one file without a `.py` extension.

timj replied:
Which is fine, but I think from what I see there that we don't then test any of the other python code. Adding a second call to flake8 is needed I think (or adding `python/` to the arguments for that flake8 call).

jhoblitt replied:
All of the python code is flake8'd by `pytest-flake8`, only the bin is being missed and that should go away if it's converted to a console script.

timj replied:
How do you explain the flake8 problems on master then?

jhoblitt replied:
flake8 wasn't complaining *a year ago* when master was last built.  flake8 output is visible in the console of the most recent build: https://travis-ci.org/lsst/lsst_build/jobs/339213143

jhoblitt committed:
DRY requirements.txt

jhoblitt committed:
tidy .travis.yml

jhoblitt committed:
add markdownlint to travis

jhoblitt committed:
add yamllint to travis

jhoblitt committed:
rm usage of builtins

jhoblitt committed:
require flake8

jhoblitt commented:
I've removed the usage of `buildtins` but something with `pytest-flake8` is now broken -- the versions in `setup.py` probably need to be reviewed but I've got to run.

timj commented:
@jhoblitt you need pytest-flake8 1.0.4 to work with the flake8 >= 3.7.5 (which is the version we use elsewhere).

jhoblitt committed:
update pypi dep versions

jhoblitt committed:
fix flake8 E117 errors

jhoblitt committed:
use an extra feature for travis

This is to avoid having flake8 version declared multiple times in
different places.

jhoblitt committed:
rm py < 3.6 from "classifiers" list

jhoblitt commented:
@timj This PR should be ready for review now.

timj commented on .mdlrc:
```diff
@@ -0,0 +1,5 @@
+# a seperate "style" file must be used to pass "parameters" to a rule
```
seperate -> separate

jhoblitt replied:
fixed.

timj commented on README.md:
```diff
@@ -0,0 +1,211 @@
+------------------
+
+`lsst-build prepare` clones the given top level product(s) into `<builddir>`,
+discovers its depenencies by reading its `ups/$PRODUCT.table` file, and
```
depenencies -> dependencies

jhoblitt replied:
fixed.

timj reviewed (approved):
Looks good to me. At some point we should consider moving much of the setup.py configuration parameters into the setup.cfg (setuptools has supported that for a long time). Thanks for the great documentation although I'm surprised it's not in reST format. Would @jonathansick prefer it in sphinx form?

jhoblitt committed:
fix speling

jhoblitt commented:
@timj I'm not the originator of the markdown -- it was the original format in the `README`. What I did was fix up some problems and get it to pass `markdownlint`.  I have no objection to it being converting to rst.  The main objective of this PR was to get the "tests" back into a working state.
